### DISCUSSION TITLE
NSUserTrackingUsageDescription Problem

### DISCUSSION THREAD

**Discussion by MESepehr** [airsdk/Adobe-Runtime-Support#1916] [General]
Hi Guys, I have a question about the app that I recently developed.
I submitted it to the Apple store and they rejected to approve it. They said you are collecting advertising data but I removed all the elements, codes, ANEs , ... that were related to advertising data.
They say you have NSUserTrackingUsageDescription on your app but also I remove this tag

```
 <iPhone>
    <InfoAdditions><![CDATA[
<key>UIPrerenderedIcon</key>
<true/>
<key>UIDeviceFamily</key>
<array>
  <string>1</string>
  <string>2</string>
</array>
<key>MinimumOSVersion</key>
<string>10.0</string>
]]></InfoAdditions>
 <Entitlements><![CDATA[
<key>application-identifier</key>
<string>DG78DDS479.com.deafaw.deafanimationworld</string>
<key>keychain-access-groups</key>
<array>
  <string>DG78DDS479.*</string>
</array>
<key>get-task-allow</key>
<false/>
<key>aps-environment</key>
<string>production</string>
]]></Entitlements>
    <requestedDisplayResolution>high</requestedDisplayResolution>
  </iPhone>
  <customUpdateUI>false</customUpdateUI>
  <allowBrowserInvocation>false</allowBrowserInvocation>
  <extensions>
    <extensionID>com.distriqt.MediaPlayer</extensionID>
    <extensionID>com.distriqt.Core</extensionID>
    <extensionID>androidx.core</extensionID>
    <extensionID>androidx.appcompat</extensionID>
    <extensionID>androidx.vectordrawable</extensionID>
    <extensionID>com.google.guava</extensionID>
     <extensionID>com.distriqt.NetworkInfo</extensionID>
     <extensionID>com.distriqt.InAppBilling</extensionID>
  </extensions>
</application>
```

Would you please help me with that?

**Comment by ajwfrost** [comment]
Hi

We don't have any instance of "NSUserTrackingUsageDescription" in our codebase .. so if this is present, it's likely to be injected by an ANE.

Can you rename your .ipa file to a .zip and expand it, then check in the Info.plist file in the subfolder structure there? Just to check what the final info.plist is looking like ... is sounds like Apple are saying that this includes that description. The value should be a string, and it may hep identify where it's coming from. It may also be worth similarly expanding the ANE files to search for the usage description key in those, I'm wondering if there are some localisation strings that may be causing the value to be injected..

thanks

**Reply by MESepehr** [reply]
Hi, thanks for your reply. Below is my info.plist. There is no NSUserTrackingUsageDescription. Would you please help me for any solution? Thx
[Info.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/8700187/Info.zip)

**Reply by MESepehr** [reply]
![NsUSer](https://user-images.githubusercontent.com/4157133/168603938-898869f4-53c5-4a52-a367-79d64983e94f.jpg)

**Reply by ajwfrost** [reply]
Hmm.... so yes, it's not in that Info.plist file..! Is it possible to do a full search within the whole IPA to see if there's a mention of this string?

If not, it may be worth seeing if you can clarify with Apple where they are finding that from! as there may be some false-positive test that they need to correct...

thanks

**Reply by marchbold** [reply]
They are likely checking the version you have live in the store. Seems like a catch-22 situation, but I'd definitely try a new build to test flight and see if you can at least get it running there. Then ask them to clarify.
